cluster/src/main/java/io/scalecube/cluster/ClusterImpl.java

```diff
@@ -351,4 +352,66 @@ public NetworkEmulator networkEmulator() {
   public boolean isShutdown() {
     return onShutdown.isDisposed();
   }
+
+  public interface ClusterMonitorMBean {
+    String getMember();
+    int getIncarnation();
+    Map<String, String> getMetadata();
+    List<String> getAliveMembers(); // all alive members except local one
+    List<String> getDeadMembers(); // let's keep recent N dead members for full-filling this method
+    List<String> getSuspectedMembers();
+  }
+
+  public static class ClusterMonitor implements ClusterMonitorMBean {
```

Comment:
Due to an issue [mentioned here](https://github.com/scalecube/scalecube-cluster/pull/174/files#r261244760)  it's clear  we can't lock up cluster  monitoring logic in a single class.
In `ClusterMonitor` and in it's XMBean counterpart, let's leave only functions: `getMember` and `getMetadata` (mbean name will stay  the same). 
Other functions concerning incarnation, membership and members must be moved to new class `MembershipProtocolMonitor` (new mbean name must be created, e.g. `"io.scalecube.cluster:name=Membership{"+localMember.id()+"}"`).